### PR TITLE
Gcc10 settable flags

### PR DESCRIPTION
**Pull request by kgerheiser**
Also remove `-free` flags from the compiler and change the file extension to `.f90` instead

### CODE REVIEW

**Review by climbfuji — APPROVED** [review]
Ok by me. I usually prefer to use uppercase filename extensions to enable automatic preprocessing when needed.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -7,16 +7,28 @@ elseif(UNIX)
 endif()
 
 if(CMAKE_C_COMPILER_ID MATCHES "^(Intel)$")
-  set(CMAKE_Fortran_FLAGS "-g -fpp -free -xHOST -traceback -FR")
-  set(fortran_8_flags "-i8 -r8")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
```
Comment (aerorahul) [inline src/CMakeLists.txt]: Don't think this is necessary.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -7,16 +7,28 @@ elseif(UNIX)
...
+  set(CMAKE_C_FLAGS_DEBUG "-g -traceback")
 elseif(CMAKE_C_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
```
Comment (aerorahul) [inline src/CMakeLists.txt]: Not necessary.

**Comment by kgerheiser** [comment]
That's what I was going to do, but then I already named the `.f90` on my Mac, which has a case-insensensitive filesystem and git wouldn't recognize the change. So, I changed it in CMakeLists.

But anyway, I went back and changed them through the Github GUI

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -7,16 +7,28 @@ elseif(UNIX)
...
   set(CMAKE_Fortran_FLAGS_DEBUG "-ggdb -Wall")
   set(fortran_8_flags "-fdefault-integer-8 -fdefault-real-8")
 endif()
 
-set(fortran_src baciof.f bafrio.f chk_endianc.f)
+if(${CMAKE_Fortran_COMPILER_ID} MATCHES "^(GNU)$" AND ${CMAKE_Fortran_COMPILER_VERSION} VERSION_GREATER_EQUAL 10)
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -w -fallow-argument-mismatch")
+endif()
+
+set(fortran_src baciof.F90 bafrio.F90 chk_endianc.F90)
```
Comment (aerorahul) [inline src/CMakeLists.txt]: Are there preprocessor directives in these?

**Review by aerorahul — CHANGES_REQUESTED** [review]
Empty CMAKE_C_FLAGS don't need to be set.  They will be taken directly from env.
See comment on preprocessor files for `.F90`

File: src/CMakeLists.txt (hunk 3 above)
Comment (kgerheiser) [reply src/CMakeLists.txt]: No

File: src/CMakeLists.txt (hunk 1 above)
Comment (kgerheiser) [reply src/CMakeLists.txt]: Right, I see that now. If nothing overwrites the flag it's fine.

File: src/CMakeLists.txt (hunk 3 above)
Comment (aerorahul) [reply src/CMakeLists.txt]: I think only `bacio.F90` has `include` and that is fine.
The other 2 can be `.f90`

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -7,16 +7,26 @@ elseif(UNIX)
 endif()
 
 if(CMAKE_C_COMPILER_ID MATCHES "^(Intel)$")
-  set(CMAKE_Fortran_FLAGS "-g -fpp -free -xHOST -traceback -FR")
-  set(fortran_8_flags "-i8 -r8")
+  set(CMAKE_C_FLAGS_DEBUG "-g -traceback")
```
Comment (aerorahul) [inline src/CMakeLists.txt]: Actually, this should be `set(CMAKE_C_FLAGS "-g -traceback ${CMAKE_C_FLAGS}")`

File: src/CMakeLists.txt (hunk 3 above)
Comment (climbfuji) [reply src/CMakeLists.txt]: We probably have diverging philosophies on this. My take is that preprocessing the file doesn't cost much and avoids potential changes in the future. But either way is ok for me.

Comment (aerorahul) [reply src/CMakeLists.txt]: - Just looking at the file extension tells us that there is preprocessor involved.
- CMake processes those files first to produce `.f90` which are then compiled.

Comment (kgerheiser) [reply src/CMakeLists.txt]: That's actually a Fortran `Include` which doesn't use the pre-processor

Comment (aerorahul) [reply src/CMakeLists.txt]: even better. remove the `-fpp` completely and turn those files to `.f90`

File: src/CMakeLists.txt [hunk 5]
```diff
@@ -7,16 +7,28 @@ elseif(UNIX)
...
 elseif(CMAKE_C_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
-  set(CMAKE_C_FLAGS_DEBUG "-ggdb -Wall")
-  set(CMAKE_Fortran_FLAGS "-g -ffree-form -fbacktrace -funroll-loops")
-  set(CMAKE_Fortran_FLAGS_DEBUG "-ggdb -Wall")
+  set(CMAKE_C_FLAGS "-g ${CMAKE_C_FLAGS}")
+  set(CMAKE_C_FLAGS_DEBUG "-Wall")
+endif()
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
+  set(CMAKE_Fortran_FLAGS "-g -fpp -traceback -FR ${CMAKE_Fortran_FLAGS}")
```
Comment (aerorahul) [inline src/CMakeLists.txt]: i think you missed a spot `-fpp`

**Review by aerorahul — APPROVED** [review]

**Pull request merged**